homeassistant/components/zha/__init__.py

```diff
@@ -285,13 +301,47 @@ def register_entity(self, ieee, entity_obj):
         discovery_info.update(entity_info)
         self._hass.data[DISCOVERY_KEY][cluster_key] = discovery_info
 
-        await discovery.async_load_platform(
-            self._hass,
-            component,
-            DOMAIN,
-            {'discovery_key': cluster_key},
-            self._config,
-        )
+        from zigpy.zcl.clusters.general import OnOff
+        if component == 'binary_sensor' and OnOff.cluster_id == \
+                cluster.cluster_id:
+            await self._async_setup_remote(discovery_info)
+        else:
+            await discovery.async_load_platform(
+                self._hass,
+                component,
+                DOMAIN,
+                {'discovery_key': cluster_key},
+                self._config,
+            )
+
+    async def _async_setup_remote(self, discovery_info):
+
+        async def safe(coro):
+            """Run coro, catching ZigBee delivery errors, and ignoring them."""
+            import zigpy.exceptions
+            try:
+                await coro
+            except zigpy.exceptions.DeliveryError as exc:
+                _LOGGER.warning("Ignoring error during setup: %s", exc)
+
+        from zigpy.zcl.clusters.general import OnOff, LevelControl
+        out_clusters = discovery_info['out_clusters']
+        if OnOff.cluster_id in out_clusters:
+            cluster = out_clusters[OnOff.cluster_id]
+            if discovery_info['new_join']:
+                await safe(cluster.bind())
+                await safe(cluster.configure_reporting(0, 0, 600, 1))
+            self._hass.data[DATA_ZHA_EVENT].append(
+                ZHASwitchEvent(self._hass, cluster, discovery_info)
+            )
+        if LevelControl.cluster_id in out_clusters:
+            cluster = out_clusters[LevelControl.cluster_id]
+            if discovery_info['new_join']:
+                await safe(cluster.bind())
+                await safe(cluster.configure_reporting(0, 1, 600, 1))
```

Comment:
I'm not sure having "min_report_interval" at such low value is a good idea. Two reasons:
1. remotes are unlikely to send attribute reports anyway, as those usually just send "cluster commands"
2. if for some reason it would send attribute reports (which I still have to see), you are risking to put too much traffic onto network. I've tried it with the lights and when dimming more then one light at the same time, those reports were overloading the network to the point it was causing Delivery Timeout exceptions.

Reply:
All of this came from the binary_sensor component of zha. It is all pre-existing logic. I’ll look into it as well though.